website/docs/tutorials/07-testing/03-fuzzing/strategies.md

```diff
@@ -0,0 +1,135 @@
+# Strategies
+
+Often in code we operate on some assumptions about the possible values of variables. 
+For example we can perform greater than zero validation early in the code, and we can assume that variables are greater than zero in the subsequent code. 
+Fuzzer allows to provide such assumptions to avoid testing against values which are not covered by the code. 
+
+Cheatcode [`given`](../02-cheatcodes/given.md) ,
+instructs fuzzer how to constraint set of values code is tested against.
+Such constraints are provided declaratively, by assigning _strategies_ to the input parameters as on the example below.
+
+```cairo title="Example"
+@external
+func setup_integers() {
+    %{
+        given(
+            a = strategy.integers(10, 20),
+            b = strategy.integers(30, 40),
+        )
+    %}
+    return ();
+}
+
+@external
+func test_integers{syscall_ptr : felt*, range_check_ptr}(a : felt, b : felt) {
+    assert_le(a, b);
+    return ();
+}
+```
+
+:::note
+By default, Protostar applies the [`strategy.felts()`](#strategyfelts) strategy to all felt
+parameters.
+:::
+
+This document is a guide to what strategies are available for generating examples and how to build
+them.
+
+## Core strategies
+
+All core strategies are contained in the [`strategy`](../02-cheatcodes/strategy.md)
+cheatcode-namespace.
+
+### `strategy.felts`
+
+```python
+def felts() -> Strategy: ...
+```
+
+Explores all possible felt values.
+
+### `strategy.integers`
+
+```python
+def integers(
+    min_value: Optional[int] = None,
+    max_value: Optional[int] = None,
+) -> Strategy: ...
+```
+
+Generates integer values, possibly bounded by provided range.
+
+```python title="Examples"
+strategy.integers(0, 100)
+strategy.integers(max_value=3000)
+strategy.integers(11)
+```
+
+Fuzzer picks integers from provided range and then converts them to felts.
+If `min_value` is not `None` then all values will be greater than or equal to `min_value`,
+and if `max_value` is not `None` then all values will be less than or equal to `max_value`.
+When applied to field elements, the unbounded values may rarely overflow.
+
+## Adapting strategies
+
+Often it is the case that a strategy does not produce exactly what is desired and a need to further
+adapt the generated values arise.
+Although this could be done in tests directly, this hurts because fuzzer does not know about the adaptation and may repeatedly test same values.
```

Comment:
```suggestion
Although this could be done in tests directly, this hurts because fuzzer does not know about the adaptation and may repeatedly test the same values.
```